packages/openneuro-components/src/dataset/DatasetTools.tsx

```diff
@@ -89,7 +90,9 @@ export const DatasetTools = ({
         </Tooltip>
       )}
       <span>
-        <Link className="dataset-tool" to={`/datasets/${datasetId}/download`}>
+        <Link
+          className="dataset-tool"
+          to={`/datasets/${datasetId}/versions/${snapshotId}/download`}>
```

Comment:
If I'm logged in and navigate to a published dataset I own it defaults to the draft view and the snapshotId is undefined, Following through with the download link the api produces a 5xx error.

Reply:
Added a fix for this with new test coverage for the issue.